api/deployment/list.go

```diff
@@ -138,7 +138,7 @@ func ListDeployments(c *gin.Context) {
 	dWithBs := []*library.Deployment{}
 
 	for _, deployment := range d {
-		b, err := database.FromContext(c).GetDeploymentBuildList(*deployment.URL)
+		b, _, err := database.FromContext(c).ListBuildsForDeployment(deployment, nil, 1, 3)
```

Comment:
I'm curious about this line here. Is there a maximum number of three builds that can result from a deployment?

Reply:
My memory is a little fuzzy on the exact details but I believe `3` was chosen as a starting point? 😅 

IIRC, we don't have anything that limits the amount of builds that can be tied to a specific deployment 👍 

The reason why I'm using `3` here is because that's how the code was originally implemented:

https://github.com/go-vela/server/pull/471

FWIW - I have no objections to increasing this to `10` since that's more consistent with our default `per_page`.